### PR TITLE
[Tests] Update ==> patch in skips.go

### PR DESCRIPTION
**Pull request by nunnatsa**
### What this PR does
Before this PR:
the `SkipIfMissingRequiredImage` and the `SkipIfNoRhelImage` function are also modifying the PV. They use `Update` for this modification.

After this PR:
Split the function into two functions. The modification function uses Patch instead of Update.

### Release note
```release-note
None
```

/sig code-quality

### CODE REVIEW

**Review by xpivarc — COMMENTED** [review]
/hold
We should remove `skips.go`

**Comment by akalenyu** [comment]
> /hold We should remove `skips.go`

There's some complexity here that we could tackle in this PR, which would help these efforts. It's just that it's going to be hard to verify (sysprep/win lane?)
For example we could use pv reclaim policy "recycle" https://kubernetes.io/docs/tasks/administer-cluster/change-pv-reclaim-policy/

**Comment by nunnatsa** [comment]
@xpivarc - this pr handles abuses of the some skip functions, by splitting the pv update out of this file. This will help to remove this file in the future, even if this is not the purpose of the pr.

/unhold

**Comment by nunnatsa** [comment]
rebased to resolve conflicts

**Review by RamLavi — APPROVED** [review]
I totally agree that this file should be gone, but this PR makes a difference in making it a little more cleaner until it is removed. 

/lgtm

File: tests/libstorage/pv.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		pv.Spec.NodeAffinity.Required != nil &&
+		len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms) > 0 &&
+		(pv.Spec.VolumeMode == nil || *pv.Spec.VolumeMode != k8sv1.PersistentVolumeBlock)
+}
+
+func isPVAvailable(pv k8sv1.PersistentVolume) bool {
+	return pv.Spec.ClaimRef == nil
+}
+
+func MakePVAvailable(ctx context.Context, pv *k8sv1.PersistentVolume) error {
```
Comment (akalenyu) [inline tests/libstorage/pv.go:25]: So we could get rid of this if we use pv reclaim policy "recycle" https://kubernetes.io/docs/tasks/administer-cluster/change-pv-reclaim-policy/ https://github.com/kubevirt/kubevirt/blob/31fa6dda29259ef93272e1b7caecc731fcf8541a/automation/test.sh#L366

**Review by akalenyu — COMMENTED** [review]
Looks good, suggestion if you want to get rid of some extra boiler plate.
Main worry is about exercising these lanes - AFAIK, these are periodics so if this PR breaks them, we wouldn't know until later

**Review by xpivarc — COMMENTED** [review]
I believe it is not productive to maintain functions/utils which should have been removed already. I see a good step forward where we split some responsibility out the Skip but I don' t see reason to avoid to do one more step where we actually remove this skip all together.

I also opened a PR to formality familiarize the removal of the skips https://github.com/kubevirt/kubevirt/pull/12534

**Pull request closed without merging**